src/server/system_services/replication_store.js

```diff
@@ -69,6 +70,37 @@ class ReplicationStore {
         });
         return ans;
     }
+
+    async update_replication_status_by_id(_id, rule_id, status) {
+        dbg.log0('update_replication_status_by_id: ', _id, rule_id, status);
+        const parsed_id = db_client.instance().parse_object_id(_id);
+
+        const bulk = this._replicationconfigs.initializeOrderedBulkOp();
```

Comment:
why do you need a bulk update here?

Reply:
as we discussed, didn't know there is a problem using bulk and conversion of queries with '.$.' was only exist in bulk so I used it, anyway I moved the conversion of .$. queries to a function in postgres client and called it also from updateOne (for non bulk)